### PR TITLE
Support compact SD and complete video diffusion workflows

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary
- export BK-SDM checkpoints without a UNet mid-block
- add encoder-only T5 export for CogVideoX
- add typed DDIM epsilon/sample/v-prediction contracts
- add model-agnostic video CFG, layout, unscale, causal-cache, and recurrence workflows
- add typed cast/clamp adapters and nested artifact paths for image APIs/image edit

## Validation
- 76 policy/workflow tests passed
- earlier targeted model tests: 27 passed
- real Stable Diffusion and CogVideoX packages validated and executed through ONNX GenAI

### CODE REVIEW

**Comment by justinchuby** [comment]
Final catalogue validation: 9bdf3150 makes image-edit output clamping cast BF16 decoder output to FP32 before Clip, avoiding ORT BF16 Less/Clip kernel assignment failure. Targeted policy/workflow tests: 77 passed. Real Qwen-Image-Edit-2509 generic workflow then passed end to end on CUDA.

File: src/mobius/generation/_policy_components.py [hunk 1]
```diff
@@ -2004,16 +2049,31 @@ def build_ddim_solver_step(
...
+    if prediction_type == "epsilon":
+        pred_original = op.Div(
+            op.Sub(sample, op.Mul(beta_sqrt, derivative)), alpha_sqrt
+        )
+        pred_epsilon = derivative
+    elif prediction_type == "sample":
+        pred_original = derivative
+        pred_epsilon = op.Div(
+            op.Sub(sample, op.Mul(alpha_sqrt, pred_original)), beta_sqrt
+        )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/generation/_policy_components.py:2233]: Fixed in 37e75275. Added nontrivial-alpha runtime parity for `prediction_type="sample"`, including epsilon reconstruction, plus a unit-alpha regression case.

File: src/mobius/integrations/onnx_genai/workflow_metadata.py [hunk 2]
```diff
@@ -3903,7 +4073,46 @@ def build_video_diffusion_workflow_metadata(
...
+                ),
+                _invoke(
+                    "guidance_combine",
+                    {
+                        "unconditional": "denoiser.unconditional",
+                        "conditional": "denoiser.conditional",
+                        "scale": "request.guidance_scale",
+                    },
+                    {"estimate": "denoiser.estimate"},
+                ),
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/onnx_genai/workflow_metadata.py:4232]: Fixed in 37e75275. Guided video metadata tests now assert negative conditioning, two denoiser invocations, and execute rank-5 CFG combination.

File: src/mobius/integrations/onnx_genai/workflow_metadata.py [hunk 3]
```diff
@@ -3623,6 +3735,8 @@ def build_video_diffusion_workflow_metadata(
     latent_permutation: list[int] | None = None,
     solver: str = "euler",
     clip_sample_range: float | None = None,
+    prediction_type: str = "epsilon",
+    guidance_scale: float | None = None,
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/onnx_genai/workflow_metadata.py:3799]: Fixed in 37e75275. Automatic export now structurally detects rank-5 denoisers, routes to the generic video workflow, and materializes faithful DDIM schedule parameters from Diffusers config.

File: src/mobius/integrations/onnx_genai/workflow_metadata.py [hunk 4]
```diff
@@ -3744,6 +3863,10 @@ def build_video_diffusion_workflow_metadata(
             "solver_step", build_euler_solver_step(sample_input.dtype, latent_dims)
         )
     pkg.add_policy_component("continue_predicate", build_boolean_not())
+    if guidance_scale is not None:
+        pkg.add_policy_component(
+            "guidance_combine", build_guidance_combine(estimate_output.dtype, latent_dims)
+        )
     pkg.add_policy_component(
         "video_latent_init",
         build_video_latent_initializer(sample_input.dtype, init_noise_sigma),
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/onnx_genai/workflow_metadata.py]: Fixed in 37e75275. The denoiser timestep dtype is now threaded into initializer history and all recurrence contracts; mixed-dtype tests assert agreement.

[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/onnx_genai/workflow_metadata.py]: Fixed in 37e75275. Latent and VAE dtypes are threaded through permutation, unscale, chunk/count, cache, conditioning, and optional denoiser-to-VAE casts, with mixed-precision coverage.

File: src/mobius/models/t5.py [hunk 5]
```diff
@@ -475,6 +475,40 @@ def preprocess_weights(
...
+        self.config = config
+        self.encoder = T5Encoder(config)
+
+    def forward(
+        self,
+        op: OpBuilder,
+        input_ids: ir.Value,
+        attention_mask: ir.Value | None = None,
+    ):
+        return self.encoder(op, input_ids=input_ids, attention_mask=attention_mask)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/t5.py:508]: Fixed in 37e75275. T5 now broadcasts the padding mask into additive attention bias, and a graph test verifies `attention_mask` is consumed.

File: src/mobius/models/t5.py [hunk 6]
```diff
@@ -475,6 +475,40 @@ def preprocess_weights(
         return new_state_dict
 
 
+class T5EncoderModel(nn.Module):
+    """Encoder-only T5 model used as a diffusion prompt encoder."""
+
+    default_task = "t5-text-encoding"
+    category = "encoder"
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/t5.py:495]: Fixed in 37e75275. `T5EncoderModel` is imported and included in the public `mobius.models` API, with an export assertion.

File: src/mobius/models/unet.py [hunk 7]
```diff
@@ -681,7 +685,8 @@ def forward(
             down_block_res_samples.extend(res_samples)
 
         # Mid
-        sample = self.mid_block(op, sample, emb, encoder_hidden_states)
+        if self.mid_block is not None:
+            sample = self.mid_block(op, sample, emb, encoder_hidden_states)
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/unet.py:689]: Fixed in 37e75275. Added compact UNet graph-build coverage for `mid_block_type=None`, exercising the skip-connection path used by BK-SDM.

**Comment by justinchuby** [comment]
Review cleanup is complete at `37e75275`. This adds the requested video CFG/runtime coverage, automatic rank-5 CogVideoX dispatch, faithful DDIM schedule materialization, typed history/layout/cache/VAE adapters, T5 padding-mask semantics/public export, compact UNet coverage, and image-edit artifact/output-clamp ordering coverage.

All 8 substantive review threads are resolved. Local targeted validation: 169 passed, 1 skipped; Ruff 0.16.2 format/lint and `git diff --check` pass. CI currently has 20 successful checks and no failures; three GPU/golden jobs remain queued. This PR has not been merged.

**Comment by justinchuby** [comment]
Rubber-duck blockers are fixed at `c84e542a`:

- Image-edit and video exports now reject omitted guidance instead of silently selecting `1.0`/unguided execution. `mobius build --runtime onnx-genai --guidance-scale SCALE` parses and forwards the explicit value; `1.0` remains an intentional unguided choice, while values above one preserve negative-conditioning CFG wiring.
- Image-edit publishes its image contract from the `image_output_clamp` output, so FP16/BF16 VAE decoders correctly declare a float32 image.

Regressions cover omitted image/video guidance, CLI forwarding, and FP16/BF16 output contracts. Targeted tests: 106 passed. Ruff 0.16.2 format/lint and `git diff --check` pass. No unresolved review threads remain; PR remains unmerged.

**Comment by justinchuby** [comment]
Final pinned-revision blocker is fixed at `5fb8cc78` (kept unmerged).

- Image-edit scheduler resolution now receives `revision`.
- Video and generic diffusion VAE scaling-factor resolution now receives `revision`; `load_diffusers_vae_scaling_factor()` pins the `vae/config.json` Hub download.
- Audit also found the standalone ComfyUI conversion scheduler lookup; its API now accepts and forwards an optional pinned revision.
- Existing tokenizer, processor/runtime-asset, diffusers component config, and weight reads already forward the revision.
- Added exact-argument regressions for image-edit scheduler, video scheduler/VAE, generic scheduler/VAE, direct VAE Hub download, ComfyUI scheduler, and CLI metadata forwarding.

Validation: targeted auto-export/inference-metadata/ComfyUI/CLI suites: `144 passed, 1 skipped`; Ruff 0.16.2 check/format and `git diff --check` pass. All review threads are resolved; CI is running on the new SHA.

**Comment by justinchuby** [comment]
Final `convert-comfyui` caller gap is fixed at `738a649a` (kept unmerged).

- Added `mobius convert-comfyui --revision REVISION` with direct help text.
- `_cmd_convert_comfyui` now forwards the exact parsed revision to `convert_comfyui_workflow`.
- Added a CLI-level regression that invokes `main()`, parses the full command, and asserts exact checkpoint/revision propagation.
- Audited command aliases/docs: there is no alias or separate checked-in command documentation; the argparse help is the authoritative directly related documentation and now includes the option.

Validation: targeted CLI/ComfyUI/auto-export tests `108 passed, 1 skipped`; Ruff check/format, `git diff --check`, and live `PYTHONPATH=src python3 -m mobius convert-comfyui --help` verification pass. All review threads are resolved; CI is running on the new SHA.

**Comment by justinchuby** [comment]
Latest `origin/main` (`96036366`) is merged without rebasing at `d57234e8`; PR remains unmerged and is now conflict-free/mergeable.

Conflict resolution preserved both sides: generalized arbitrary-rank typed output clamping (including float32 image-edit clamp output), main’s default rank-4 clamp API, explicit image/video guidance, canonical image value ranges, complete pinned revision propagation including `convert-comfyui --revision`, video/image workflow metadata, and all current-main conformance tests. Unrelated GGUF/SenseNova main changes were retained.

Validation: focused CLI/ComfyUI/auto-export/inference/workflow/policy/value-range set `235 passed, 1 skipped`; GGUF/SenseNova conflict areas `53 passed`; Ruff 0.16.2 check/format and diff checks pass. Worktree is clean, all review threads resolved, CI is running on `d57234e8`.

**Comment by justinchuby** [comment]
Branch restoration is pushed at `edf41ee9` without force-pushing; PR remains unmerged and mergeable. The displaced head is preserved remotely as `backup/catalogue-diffusion-builds-789bd76c`.

Root cause evidence: GitHub recorded a `head_ref_force_pushed` event at 2026-08-22 15:39:23Z by `justinchuby`. The new history was linearized onto main #541/#543 and ended in `789bd76c`, whose metadata identifies Copilot session `39ff6824-d35f-4d3b-8f5f-043a7119a100`. All six matrix logs had the identical failure: stale `test_ort_genai_runtime_is_rejected_before_artifacts` expected rejection, while #543 intentionally supports and dispatches `ort-genai`, so execution reached the nonexistent GGUF file.

Restoration: merged current `origin/main` `c047f77f` into the preserved rewritten head, retaining it as an ancestor. Updated only the contradictory GGUF test to assert exact `runtime="ort-genai", save_model=False` forwarding. Approved diffusion core blobs for `auto_export.py`, `convert.py`, and `tests/cli_test.py` exactly match `d57234e8`; workflow differences are current-main unrelated/full-duplex changes or formatting, while explicit guidance, float32 clamp output, revision propagation, and canonical image/video metadata remain intact.

Validation: GGUF suites `379 passed`; focused diffusion/CLI/ComfyUI/metadata/policy suite `239 passed, 1 skipped`; Ruff and diff checks pass. Full local unit run reached `6269 passed, 346 skipped, 62 xfailed`; two unrelated local-only failures remained (installed diffusers API mismatch in Qwen rotary test and nondeterministic ModernBERT synthetic parity). CI is now running on `edf41ee9`.

**Comment by justinchuby** [comment]
CI update for `edf41ee9`: all six previously failing Python matrix jobs now pass (Ubuntu/Windows × Python 3.11/3.12/3.13). Lint, Architecture Diff, CodeQL, benchmark comparison, build, L1 smoke, and L3 synthetic parity also pass. Remaining jobs are queued integration/L4/L5 gates; no failure is currently reported.
